### ISSUE TITLE
Focus doesn't work correctly with nested containers

### ISSUE THREAD

**Issue by igordmn** [JetBrains/compose-jb#2172]
Compose [1.2.0-alpha01-dev741](https://github.com/JetBrains/compose-jb/releases/tag/v1.2.0-alpha01-dev741)

Failed tests: [test 1](https://teamcity.jetbrains.com/buildConfiguration/JetBrainsPublicProjects_Compose_PublishFast_1ComposeDesktopTest/3890769?buildTab=tests&status=muted&expandedTest=build%3A%28id%3A3890769%29%2Cid%3A78309), [test 2](https://teamcity.jetbrains.com/buildConfiguration/JetBrainsPublicProjects_Compose_PublishFast_1ComposeDesktopTest/3890769?buildTab=tests&status=muted&expandedTest=build%3A%28id%3A3890769%29%2Cid%3A78317)

Reproducer:
```
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material.Button
import androidx.compose.material.MaterialTheme
import androidx.compose.ui.Modifier
import androidx.compose.ui.focus.FocusRequester
import androidx.compose.ui.window.singleWindowApplication

fun main() = singleWindowApplication {
    MaterialTheme {
        Column(Modifier.fillMaxSize()) {
            Button({}) {}
            Button({}) {}
            Button({}) {}
            Box {
                Button({}) {}
            }
        }
    }
}
```

https://user-images.githubusercontent.com/5963351/177622273-7e6d4d0d-8197-46d0-b968-d49d02dd2684.mov

(on the video I only press Tab, not Shift-Tab)

**Comment by igordmn** [comment]
After reverting [this commit](https://github.com/JetBrains/androidx/commit/b1fbc771c5675eed692fd69fda1653f5147f9594), everything is fine

**Comment by igordmn** [comment]
[The issue in AOSP](https://partnerissuetracker.corp.google.com/issues/238210250)

**Comment by igordmn** [comment]
Fixed in 1.2.0-beta02

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-2172">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
